fix: code block overflow

DhavalBhimani44 opened the pull request:
# Fix code block overflow in component previews #49 

## Changes
- Added `overflow-hidden` and `max-w-full` to the preview container div to prevent horizontal overflow
- Added `overflow-auto` to the code preview div to enable scrolling for wide code blocks
- This provides a generalized solution that works for all component previews without requiring changes to individual components

## Testing
- Verified that the checkbox demo component with long class names no longer causes horizontal overflow
- Confirmed that the code preview still allows scrolling when code is wider than the container
- Tested other component previews to ensure the changes don't negatively impact their appearance

## Screenshots
Before: 
![image](https://github.com/user-attachments/assets/9233337f-0bdb-4d85-8a28-9f941d0dc40c)

After: 
![image](https://github.com/user-attachments/assets/5bb2db52-4ef8-4942-926e-d09a2cbf1756)

stabldev commented:
hi @DhavalBhimani44, I did a local test and its good.
now I was thinking, what about adding https://www.npmjs.com/package/tailwind-scrollbar with some custom styles?
like https://ui.shadcn.com/docs/components/scroll-area bar and thumb.
we can create our-own scroll-area comp. later, but for now- can you do this?

stabldev commented:
@DhavalBhimani44, take a look at this, adding grid to parent div was the mistake.
this makes it scroll on overflow-x and expand on y, see if you can implement tailwind-scrollbar
```tsx
<div className="bg-secondary/25 relative aspect-video gap-4">
  {activeTab === "preview" ? (
    <div className="not-prose grid size-full place-items-center">
      {preview}
    </div>
  ) : (
    <>
      <CopyButton value={code} />
      <div dangerouslySetInnerHTML={{ __html: highlightedCode }} />
    </>
  )}
</div>
```

DhavalBhimani44 commented:
I think I messed up, this was my first time contributing to open source, I'm not sure what to do :(. What I did:
1. I forked this repo in my github and cloned
2. Created a branch (which got merged)
3. Created second branch on which I was working
4. First branch got merged in main
5. I added main repo as upstream
6. I fetched upstream and checked out main
7. Merged upstream with main
8. I pushed fork’s main branch so that I'm up-to-date
9. I again went to the second branch fix/code-bock-overflow
10. I rebased it with main but then issues came. 

Can you help me with this please?
